client/block-builder/src/lib.rs

```diff
@@ -121,11 +121,23 @@ where
 			backend,
 		})
 	}
-
+	
 	/// Push onto the block's list of extrinsics.
 	///
-	/// This will ensure the extrinsic can be validly executed (by executing it);
+	/// This will ensure the extrinsic can be validly executed (by executing it).
 	pub fn push(&mut self, xt: <Block as BlockT>::Extrinsic) -> Result<(), ApiErrorFor<A, Block>> {
+		self.push_internal(xt, false)
+	}
+
+	/// Push onto the block's list of extrinsics.
+	///
+	/// This will treat incoming extrinsic `xt` as untrusted and perform additional checks
+	/// (currenty checking signature).
+	pub fn push_untrusted(&mut self, xt: <Block as BlockT>::Extrinsic) -> Result<(), ApiErrorFor<A, Block>> {
```

Comment:
Why do we switch the behavior here? I think it should be also called `push` and `push_trusted`.

Reply:
Thought `push` would be defaulted to `_trusted`!
Changed